### PR TITLE
DDP-5932:  Fix error with getting default/temp address

### PR DESCRIPTION
**Pull request by alextanlan**
- Fixed errors connected with getting either a default or a temp address for Medical release activity
- Changed console.error hadling (and reporting to Stackdriver) in order to have more meaningful messages about the errors

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 1]
```diff
@@ -220,442 +223,447 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
...
+            distinctUntilChanged()
+        );
+
+        const initializeStateAction$ = this.state$.pipe(
+            take(1),
+            tap(() => busyCounter$.next(1)),
+            // let's disable form while we try to load initial address
+            tap(() => this.stateUpdates$.next({isTemporarilyDisabled: true})),
+            mergeMap((state) => this.addressService.findDefaultAddress().pipe(
+                catchError(error => {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts:270]: added catchError.

Other changes in the component are only for fixing the indent.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 2]
```diff
@@ -220,442 +223,447 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
...
+                map(defaultAddress => [state, defaultAddress]),
+            )),
+            tap(([_, defaultAddress]: [ComponentState, Address | null]) =>
+                defaultAddress && this.inputAddress$.next(defaultAddress)),
+            take(1),
+            // filter for case where we need to go on to look for a temp address?
+            filter(([state, defaultAddress]) => !defaultAddress && !!(state as ComponentState).activityInstanceGuid),
+            // map(([state, _]) => state as ComponentState),
+            mergeMap(([state, _]) => this.addressService.getTempAddress(state.activityInstanceGuid).pipe(
+                catchError(error => {
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts:283]: added catchError.

Other changes in the component are only for fixing the indent.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 3]
```diff
@@ -220,442 +223,447 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
...
+                    this.logger.logWarning(this.LOG_SOURCE, `An error occurred during getTempAddress: ${error.code}, ${error.message}`);
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: A question:
should we report about these errors (no default/temp address) to GCP  
or it is just enough to write a warning to console ?

**Review by aglushko-broad — DISMISSED** [review]

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts (hunk 3 above)
Comment (MocanaAtBroad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: In these 2 cases here, I think the right think to do is to capture (like you are doing), and then log as as debug (logDebug , I believe)?
I think our base code that does the http calls might be a little too creative regarding capturing certain errors, and a bunch of other stuff, but regardless the two "not founds" are not really exceptions; client just checks if an address exists and the answer, sometimes is, it does not (404).

**Review by aglushko-broad — APPROVED** [review]

**Pull request merged**